hpolib/abstract_benchmark.py

```diff
@@ -114,7 +120,23 @@ def wrapper(self, configuration: Union[np.ndarray, ConfigSpace.Configuration, Di
                                 f'was {type(configuration)}')
 
             self.configuration_space.check_configuration(config)
-            return foo(self, configuration, **kwargs)
+
+            # Check the fidelities, if any
+            fidelity = self.opt_fidelity.get_dictionary()
+            if kwargs:
+                # If kwargs contain any fidelity parameters, take them out
+                given_fidelity = {k: kwargs.pop(k, v) for k, v in fidelity.items()}
+
+                # Ensure that the extracted fidelity values play well with the defined fidelity space
```

Comment:
I would recommend the following:

1) Get the default configuration def from the fidelity space
2) Convert it to a dict
3) Convert the given fidelity to a dict, given_def or initialize as empty dict
4) Update def with values from given_def
5) Check whether it is a valid configuration given fidelity space

Reply:
Also,

I would suggest to split it into a second wrapper function. So a single wrapper checking if the configuration is correct and one if the fidelity space is correct.

Reply:
@KEggensperger If you check the definition of the opt_fidelity property, you will notice that the workflow here is essentially the same but more condensed. This workflow also takes into account the scenario where an objective function may be passed some keyword arguments (such as 'reset' from NASCifar10BaseBenchmark) that are not fidelity parameters. If no keyword arguments are given, it automatically sends the default fidelity forward, so the objective function can potentially be defined without optional fidelity arguments, although that's probably much less user-friendly.

@PhMueller I was wondering about that as well. I'll make the change.

Reply:
1) Ah, I see, sorry I didn't look at this long enough. Could you rephrase the comment in l. 140 to say something along the lines of "for each fidelity use the given value in kwargs (and take them out) or its default value if it is not part of kwargs"
2) Yes, making this a separate function is definitely helpful